server/tserver/src/main/java/org/apache/accumulo/tserver/log/RecoveryLogsIterator.java

```diff
@@ -77,7 +79,7 @@ public RecoveryLogsIterator(ServerContext context, List<Path> recoveryLogDirs, L
 
     for (Path logDir : recoveryLogDirs) {
       LOG.debug("Opening recovery log dir {}", logDir.getName());
-      List<Path> logFiles = getFiles(vm, logDir);
+      SortedSet<Path> logFiles = getFiles(vm, logDir);
```

Comment:
I'm not sure the full implications here, but I would have assumed insert order, as they appear in the metadata, would be more important than ordering by path, which could vary due to hostname differences on the HDFS volume. If the files have unique 1-up sequence based names, you could use a custom comparator to order them that way in a set.

Reply:
From what I can tell, the naming convention for the files is [here](https://github.com/apache/accumulo/blob/main/server/tserver/src/main/java/org/apache/accumulo/tserver/log/LogSorter.java#L253).

Reply:
@ctubbsii , @dlmarion , would a change such as the following address your concerns?

Replace:

    SortedSet logFiles = new TreeSet<>();

within getFiles with:

    // Path::getName compares the last component of each Path value. In this case, the last
    // component should always have the format 'part-r-XXXXX.rf', where XXXXX are one-up values.
    SortedSet logFiles = new TreeSet<>(Comparator.comparing(Path::getName));

Therefore, only comparing the 'part-r-xxxxx.rf' portion of the filenames.

Reply:
yeah, that seem sane